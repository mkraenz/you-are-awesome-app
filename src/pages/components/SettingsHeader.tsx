import React, { FC } from "react";
import { Header, Icon } from "react-native-elements";
import { HeaderTitle } from "react-navigation-stack";
import { INavigationProps } from "../INavigationProps";
import { styles } from "../Styles";

const SettingsHeader: FC<INavigationProps> = props => (
    <Header
        leftComponent={
            <Icon
                name="arrow-back"
                onPress={() => props.navigation.goBack()}
                color={styles.header.color}
            />
        }
        centerComponent={
            <HeaderTitle style={styles.header}>Settings</HeaderTitle>
        }
    ></Header>
);

export default SettingsHeader;
