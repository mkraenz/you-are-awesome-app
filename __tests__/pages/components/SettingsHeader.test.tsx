import React from "react";
import "react-native";
// Note: test renderer must be required after react-native.
import renderer from "react-test-renderer";
import SettingsHeader from "../../../src/pages/components/SettingsHeader";

it("renders correctly", () => {
    const tree = renderer
        .create(<SettingsHeader navigation={{} as any} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
