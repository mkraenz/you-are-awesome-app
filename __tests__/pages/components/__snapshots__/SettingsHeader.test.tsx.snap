// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#2089dc",
      "borderBottomColor": "#f2f2f2",
      "borderBottomWidth": 0.5,
      "flexDirection": "row",
      "height": 64,
      "justifyContent": "space-between",
      "paddingHorizontal": 10,
      "paddingTop": 20,
    }
  }
>
  <Image
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": 64,
          "width": undefined,
        },
        undefined,
      ]
    }
    testID="headerContainer"
  />
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flex": 1,
      }
    }
  >
    <View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
          }
        }
      >
        <Text />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 3,
      }
    }
  >
    <Text
      accessibilityTraits="header"
      numberOfLines={1}
      style={
        Object {
          "color": "white",
          "fontSize": 17,
          "fontWeight": "600",
          "marginHorizontal": 16,
        }
      }
    >
      Settings
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "flex": 1,
      }
    }
  />
</View>
`;
